src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -1694,4 +1694,172 @@ public void testUpdateLocatorsCopyUnsupported() throws SQLException {
+    String schemaPattern = schema.substring(0, 1).concat("%");
+    final String table1 = "T1";
+    final String table2 = "T2";
+
+    connection
+        .createStatement()
+        .execute("create or replace table " + table1 + "(C1 int primary key, C2 string)");
+    connection
+        .createStatement()
+        .execute(
+            "create or replace table "
+                + table2
+                + "(C1 int primary key, C2 string, C3 int references "
+                + table1
+                + ")");
+
+    DatabaseMetaData dbmd = connection.getMetaData();
+
+    // Should return result for matching schema and table name
+    assertEquals(1, getSizeOfResultSet(dbmd.getPrimaryKeys(database, schema, table1)));
+
+    // Should return a result if we try a pattern match on the schema
+    assertTrue(0 < getSizeOfResultSet(dbmd.getPrimaryKeys(database, schemaPattern, table1)));
+
+    // Should return a result if we try a pattern match on the table name
+    assertTrue(0 < getSizeOfResultSet(dbmd.getPrimaryKeys(database, null, "T%")));
+
+    // Should return result for matching schema and table name
+    assertEquals(1, getSizeOfResultSet(dbmd.getImportedKeys(database, schema, table2)));
+
+    // Should return a result if we try a pattern match on the schema
+    assertTrue(0 < getSizeOfResultSet(dbmd.getImportedKeys(database, schemaPattern, table2)));
+
+    // Should return a result if we try a pattern match on the table name
+    assertTrue(0 < getSizeOfResultSet(dbmd.getImportedKeys(database, null, "T%")));
+
+    // Should return result for matching schema and table name
+    assertEquals(1, getSizeOfResultSet(dbmd.getExportedKeys(database, schema, table1)));
+
+    // Should return a result if we try a pattern match on the schema
+    assertTrue(0 < getSizeOfResultSet(dbmd.getExportedKeys(database, schemaPattern, table1)));
+
+    // Should return a result if we try a pattern match on the table name
+    assertTrue(0 < getSizeOfResultSet(dbmd.getExportedKeys(database, null, "T%")));
+
+    // Should return result for matching schema and table name
+    assertEquals(
+        1,
+        getSizeOfResultSet(
+            dbmd.getCrossReference(database, schema, table1, database, schema, table2)));
+
+    // Should return a result if we try a pattern match on any of the table or schema names
+    assertTrue(
+        0
+            < getSizeOfResultSet(
+                dbmd.getCrossReference(database, schemaPattern, table1, database, schema, table2)));
+    assertTrue(
+        0
+            < getSizeOfResultSet(
+                dbmd.getCrossReference(database, schema, table1, database, schemaPattern, table2)));
+    assertTrue(
+        0
+            < getSizeOfResultSet(
+                dbmd.getCrossReference(database, null, "T%", database, schema, table2)));
+    assertTrue(
+        0
+            < getSizeOfResultSet(
+                dbmd.getCrossReference(database, schema, table1, database, null, "T%")));
+  }
```

Comment:
there are another methods in this class that use changed methods - could you look at them and add new version that sets the new parameter?